app/routes/_index.tsx

```diff
@@ -101,11 +94,11 @@ export default function Index() {
               Is public
             </Label>
           </div>
+
           <Button 
-            type="submit" 
             className="w-full bg-blue-600 text-white py-2 rounded hover:bg-blue-700 transition duration-300"
           >
-            Save
+            {nav.state === "submitting" ? "Saving..." : "Save"}
```

Comment:
remix submits forms async behind scenes for speed etc, but you can still plug into the lifecycle